### ISSUE TITLE
Customize Logging Messages

### ISSUE THREAD

**Issue by apoc4lyps** [PowershellFrameworkCollective/psframework#574]
Hi, is it possible to customize the messages on the Console?

![image](https://user-images.githubusercontent.com/10954235/228516088-b1310236-a7b5-484f-b882-66d178b2724d.png)

Regards

**Comment by FriedrichWeinmann** [comment]
Heya, it is indeed possible to customize the messages, technically even to make them look the way you want there, but not in a practical manner :(

You can change a few things using configuration:

```powershell
Get-PSFConfig PSFramework.Message.Style.*
```

But that's mostly for disabling timestamp and function name, the color or enabling the multiline breadcrumb style. The prefix options place it at the wrong end and do not include proper colors.

The "technically" basically can be done by ...

+ Disabling timestamp and command name
+ Manually building the entire, colored message

```powershell
Set-PSFConfig PSFramework.Message.Style.FunctionName $false
Set-PSFConfig PSFramework.Message.Style.Timestamp $false

Write-PSFMessage -Level Host -Message "[<c='sub'>{0:HH:mm:ss}</c>] [<c='sub'>{1}</c>] [<c='sub'>{2}</c>] <c='red'>{3}</c>" -Format (Get-Date), '<ScriptBlock>', 'Error', 'a test message with a severity Error'
```

I really don't think you want to do that for every single message^^
That being said, I can add more customization options to the message display, like color filters or adding the level to the message or ordering the tags.
At the end of the day, so long as I only do it for messages printed on screen, I don't have to care too much about the performance cost (per-message performance overhead is a major concern of mine, as it easily scales for users with a high message rate).

**Comment by apoc4lyps** [comment]
Hi Fred, thanks I will give this a try :)

**Comment by FriedrichWeinmann** [comment]
Implemented in 1.8.289
Try out Register-PSFMessageColorTransform :)
There is also a configuration option now to include the level in the messages
